inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java

```diff
@@ -0,0 +1,181 @@
+            return;
+        }
+        LOG.info("get new SortTaskConfig:taskName:{}:config:{}", taskName,
+                JSON.toJSONString(newSortTaskConfig));
+        this.sortTaskConfig = newSortTaskConfig;
+        this.sinkContext = new Context(this.sortTaskConfig.getSinkParams());
+        this.reloadIdParams();
+        this.reloadClients();
+        // todo get IEvent2LogItemHandler
+    }
+
+    /**
+     * Reload id params.
+     */
+    private void reloadIdParams() {
+        List<Map<String, String>> idList = this.sortTaskConfig.getIdParams();
+        Map<String, ClsIdConfig> newIdConfigMap = new ConcurrentHashMap<>();
+        for (Map<String, String> idParam : idList) {
+            String inlongGroupId = idParam.get(Constants.INLONG_GROUP_ID);
+            String inlongStreamId = idParam.get(Constants.INLONG_STREAM_ID);
+            String uid = InlongId.generateUid(inlongGroupId, inlongStreamId);
+            String jsonIdConfig = JSON.toJSONString(idParam);
+            ClsIdConfig idConfig = JSON.parseObject(jsonIdConfig, ClsIdConfig.class);
+            newIdConfigMap.put(uid, idConfig);
+        }
+        this.idConfigMap = newIdConfigMap;
+    }
+
+    /**
+     * Close expire clients and start new clients.
+     * <p>Each client response for data of one uid.</p>
+     * <p>First, find all UIDs that are in the active clientMap but not in the updated id config (or to say EXPIRE UID),
+     * and put those clients into deletingClientsMap.
+     * The real close process will be done at the beginning of next period of reloading.
+     * Second, find all UIDs that in the updated id config but not in the active clientMap(or to say NEW UID),
+     * and start new clients for these UIDs and put them into the active clientMap.</p>
+     */
+    private void reloadClients() {
+        clientMap.keySet()
+                .stream()
+                .filter(uid -> !idConfigMap.containsKey(uid))
+                .forEach(this::removeExpireClient);
+        idConfigMap.keySet()
+                .stream()
+                .filter(uid -> !clientMap.containsKey(uid))
+                .forEach(this::startNewClient);
+    }
+
+    /**
+     * Start new cls client and put it to the active clientMap.
+     *
+     * @param uid UID of new client.
+     */
+    private void startNewClient(String uid) {
+        ClsIdConfig idConfig = idConfigMap.get(uid);
+        if (idConfig == null) {
+            LOG.error("Start client failed, there is not cls config of {}", uid);
+            return;
+        }
+        AsyncProducerConfig producerConfig = new AsyncProducerConfig(
+                idConfig.getEndpoint(),
+                idConfig.getSecretId(),
+                idConfig.getSecretKey(),
+                NetworkUtils.getLocalMachineIP());
+        // todo set other configs
+        AsyncProducerClient client = new AsyncProducerClient(producerConfig);
+        clientMap.put(uid, client);
```

Comment:
It is better to create AsyncProducerClient by endpoint, not to create AsyncProducerClient by id.

Reply:
The init process of AsyncProducerClient requires two important parameters, secretId and secretKey. 
Each id has unique secretId and secretKey but may share the same endpoint. 
Hence we have to new one client for each task, although it cost more resouces, it's the price that we hand over the controll of dispatch and fail retry to CLS SDK.

Reply:
details can be seen at  https://github.com/TencentCloud/tencentcloud-cls-sdk-java